packages/@okta/vuepress-theme-prose/global-components/SignUp.vue

```diff
@@ -324,7 +334,44 @@ export default {
           },
         };
 
-        this.apiService.post(registrationPath, { body })
+        this.isPending = true;
+
+        this.apiService
+          .post(registrationPath, { body })
+          .then(res => {
+            // Reset error in case of transient failure that succeeds later
+            this.error = null;
+            // Redirect user to success landing page
+            window.location.assign('/signup/thank-you');
+          })
+          .catch(err => {
+            this.handleApiError(err);
+          })
+          .finally(() => {
+            this.isPending = false;
+          });
+      };
+    },
+
+    handleApiError(err) {
+      if (err.response) {
+        const { status, data } = err.response;
+
+        switch (status) {
+          case 400: {
```

Comment:
why only 400? how difference of response when other type of error code

Reply:
I think right now `400` is the only code the user can reliably do anything about since they are usually related to a bad input. `500` and other codes are out of the user control so we just show the generic message and hope it was a transient failure, have them try again.

You can see the generic failure message in the screenshot above

Reply:
I agree with Shaw's approach here, this looks like that is the only relevant error for a user right now. Getting these changes as that is miles better than the current behaviour.